networks/rpc/server.go

```diff
@@ -45,13 +46,33 @@ const (
 
 	// pendingRequestLimit is a limit for concurrent RPC method calls
 	pendingRequestLimit = 200000
+
+	// concurrencyLimit is a limit for the number of concurrency connection for RPC servers
+	concurrencyLimit = 3000
+
+	// minFDLimit is the minimum file descriptor limit for RPC servers (heuristic value. concurrency limit of RPC and WS servers)
+	minFDLimit = concurrencyLimit * 2
 )
 
 // pendingRequestCount is a total number of concurrent RPC method calls
 var pendingRequestCount int64 = 0
 
 // NewServer will create a new server instance with no registered handlers.
 func NewServer() *Server {
+	// increase the file descriptor limit if the process has a small limit
+	limit, err := fdlimit.Current()
+	if err != nil {
+		logger.Debug("fail to read fd limit", "err", err)
+	} else {
+		if limit < minFDLimit {
+			err := fdlimit.Raise(uint64(limit + minFDLimit))
```

Comment:
To raise FD, is there any permission is needed?

Reply:
Not additional permission is needed. It changes fd limit of the process but doesn't change fd limit of the operating system.